src/platform/logging/BUILD.gn

```diff
@@ -0,0 +1,84 @@
+# See Project CHIP LICENSE file for licensing information.
+
+import("//build_overrides/build.gni")
+import("//build_overrides/chip.gni")
+import("//build_overrides/pigweed.gni")
+
+import("${chip_root}/src/platform/device.gni")
+
+config("includes") {
+  include_dirs = [ "include" ]
+}
+
+if (chip_enable_openthread) {
+  import("//build_overrides/openthread.gni")
+
+  if (chip_device_platform == "linux" || chip_device_platform == "Darwin") {
+    import("//build_overrides/ot_br_posix.gni")
+  }
+}
+
+simple_log_platforms = [
+  "cc13x2_26x2",
+  "darwin",
+  "efr32",
+  "esp32",
+  "k32w",
+  "linux",
+  "nrfconnect",
+  "qpg6100",
+]
+
+foreach(platform, simple_log_platforms) {
+  if (chip_device_platform == platform) {
+    static_library(platform) {
+      sources = [ "impl/" + platform + "/Logging.cpp" ]
+
+      deps = [
+        "${chip_root}/src/lib/core:chip_config_header",
+        "${chip_root}/src/platform:platform_buildconfig",
+      ]
+
+      public_configs = [ ":includes" ]
+
+      if (chip_enable_openthread) {
```

Comment:
aside: This is another suspicious bit of coupling..

Reply:
The logging logic is odd. Logging.cpp files define also openthread logging redirects, which need openthread includes.

The couplinig is OpenThread requiring otPlatLog

Reply:
Shouldn't it be the other way around? If `CHIP_ENABLE_OPENTHREAD == 1`  then `Logging.cpp` should provide the otPlatLog function?

Reply:
That code is already available. The problem is that otPlatformLog requires `#defines` that are in the openthread includes, so I need to have the include paths for openthread in the include path. This is what this code is doing:
   - if openthread is enabled, add a dependency to openthread to get includes
   
It does seem a bit awkard to me. Maybe we should split the logging, so I can depend on ':openthread-logging' to make things clearer.